### PR TITLE
Add fonts with broad glyph support for PDF export

### PR DESCRIPTION
**Pull request by sswatson**
I wasn't able to build this on my machine to test these additions (for reasons that were unrelated to the changes), but I did test them on top of the minimal-jupyter image, and it worked great. The reason that the original scripts didn't work is that I forgot that `mktexlsr` has to be run (as root) for TeX to recognize the system fonts.

### CODE REVIEW

**Review by mirestrepo — COMMENTED** [review]
I can't install fonts in home. Home gets overwritten by GCP persistent volume. If I install in /usr/local/share/fonts, it complaints with this error:

(/usr/share/texlive/texmf-dist/tex/latex/psnfss/mathpazo.sty)kpathsea:make_tex: Invalid fontname `Noto Sans Mono', contains ' '
    
    
    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    !
    ! fontspec error: "font-not-found"
    ! 
    ! The font "Noto Sans Mono" cannot be found.
    ! 
    ! See the fontspec documentation for further information.
    ! 
    ! For immediate help type H <return>.
    !...............................................  
                                                      
    l.7     \setmonofont{Noto Sans Mono}
                                        
    ? 
    ! Emergency stop.
    !...............................................  
                                                      
    l.7     \setmonofont{Noto Sans Mono}

**Comment by mirestrepo** [comment]
Hi Sam - I have added the changes to the docker file for datasci-data instead of here, in the interest of time so the build doesn't take so long and uses the previous cache layer. My last two commit will reveal the changes

**Comment by mirestrepo** [comment]
Okay - I figured it out. I guess it needs to be installed in a directory tree /usr/local/share/fonts/opentype/noto.

Update to docker image coming soon

**Comment by mirestrepo** [comment]
Actually, that didn't work - mathpazo still doesn't find the font in /usr/local/share/fonts

**Comment by sswatson** [comment]
I see the issue, and the fix will be easy since since the Pagella font already works (yay!). 

TL;DR tack the Noto fonts onto the layer where the other fonts are successfully installed (and remove lines 68 to 76 in the `datasci-data` Dockerfile): 

```
# Install Noto Sans Mono and TeX Gyre Pagella fonts on the system
USER root
RUN cd /usr/share/fonts && \
    wget http://mirrors.ctan.org/fonts/tex-gyre/opentype/texgyrepagella-bold.otf && \
    wget http://mirrors.ctan.org/fonts/tex-gyre/opentype/texgyrepagella-bolditalic.otf && \
    wget http://mirrors.ctan.org/fonts/tex-gyre/opentype/texgyrepagella-italic.otf && \
    wget http://mirrors.ctan.org/fonts/tex-gyre/opentype/texgyrepagella-regular.otf && \
    wget https://noto-website-2.storage.googleapis.com/pkgs/NotoSansMono-unhinted.zip && \
    unzip NotoSansMono-unhinted.zip
```

Longer version, for the record: there are two ways of getting a font into a TeX document: a TeX package (for example, the `noto` package at the CTAN url currently in the Dockerfile), and querying the OS for system fonts. The former would involve `\usepackage{noto}` in the TeX preamble, while the latter uses`\usepackage{fontspec} \setmainfont{Noto Sans Mono}` and only works with `xelatex` or `lualatex`, not `pdflatex` (but that's OK, since Jupyter uses `xelatex` by default). The `mathpazo` package is only responsible for loading the Palatino-compatible math fonts in our case (by default it also loads a Palatino main font, but we're overriding that with `\setmainfont{TeX Gyre Pagella}`).

The Dockerfile code I proposed was pursuing the latter approach, since it tends to be easier and more flexible. If we were going to use `noto.zip`, we would probably want to do that through the TeXLive manager: `tlmgr install noto` (although `tlmgr` is throwing an error for me for some reason---even on the minimal notebook and even as root. Not worth troubleshooting). That would also require a change to the template preamble. In any case, my understanding is that it's expected behavior for the `\setmonofont` command to not see fonts installed into the TeX tree.

**Comment by sswatson** [comment]
Also, I take your point about installing in `/home`. I thought that `fc-cache` was copying the files in a central location (that's how it works on macOS), but apparently it just indexes them.

**Comment by mirestrepo** [comment]
Your suggestion was the first thing I tried, but didn't work. Then because of my lack of understanding of the two methods that you just explain, I went down a rabbit hole :). I tried a million things, nothing worked. With a new fresh mind after a beach day - I just now noticed that the Noto fonts default permissions were 440, therefore Joyvan couldn't read and that iswhy Pangella was working but Noto was not! So mystery solved. Update coming soon, I'll push this update, then I'll take the image resizing PR

**Comment by mirestrepo** [comment]
Image has been pushed to hub! Wohoo

**Comment by sswatson** [comment]
🎉📣🎉
